Panic when a contract is not renewed

When a contract expires and is not renewed in a due time (for whatever reason), it stays in the list of active contracts until the current block height passes the end height of the contract, and then a panic is triggered:
```go
panic: current blockHeight should be lower than the endHeight
goroutine 61 [running]:
go.sia.tech/core/rhp/v2.ContractRenewalCollateral({0x2c000000, {0xd0, 0xda, 0xe, 0xc3, 0x0, 0xd1, 0x2d, 0xba, 0x61, ...}, ...}, ...)
	.../go/pkg/mod/go.sia.tech/core@v0.1.11-0.20230324142811-50da7dba9eb7/rhp/v2/contracts.go:87 +0x311
go.sia.tech/renterd/autopilot.isOutOfCollateral({{{0x3d, 0x30, 0x2f, 0x65, 0x45, 0xed, 0x46, 0x7, 0x8a, 0xec, ...}, ...}, ...}, ...)
	.../renterd/autopilot/hostfilter.go:244 +0x145
go.sia.tech/renterd/autopilot.isUsableContract({{0x0}, {0x0, 0x150, 0x0}, {{0xc000aa1a30, 0x9}, 0x32, {0x9fd0803ce8000000, 0x33b2e3c}, 0x3f0, ...}}, ...)
	.../renterd/autopilot/hostfilter.go:198 +0x107
go.sia.tech/renterd/autopilot.(*contractor).runContractChecks(0xc000ad6140, {0x15714b0, 0xc0004b7110}, {0x1577288, 0xc000ac1890}, {0xc0001e8800, 0x2, 0x4}, 0x0?)
	.../renterd/autopilot/contractor.go:488 +0xf25
go.sia.tech/renterd/autopilot.(*contractor).performContractMaintenance(0xc000ad6140, {0x15714b0?, 0xc0004b6150?}, {0x1577288, 0xc000ac1890})
	.../renterd/autopilot/contractor.go:269 +0xbf6
.................................
I think the contract should be removed from the active contracts if it is not renewed.

--- a/autopilot/contractor.go
+++ b/autopilot/contractor.go
@@ -494,7 +494,9 @@ func (c *contractor) runContractChecks(ctx context.Context, w Worker, contracts 
 				"renew", renew,
 			)
 		}
-		if renew {
+		if archive {
+			toArchive[fcid] = errStr(joinErrors(reasons))
+		} else if renew {
 			renewIndices[fcid] = len(toRenew)
 			toRenew = append(toRenew, contractInfo{
 				contract: contract,
@@ -505,8 +507,6 @@ func (c *contractor) runContractChecks(ctx context.Context, w Worker, contracts 
 				contract: contract,
 				settings: settings,
 			})
-		} else if archive {
-			toArchive[fcid] = errStr(joinErrors(reasons))
 		}
 
 		// keep track of file size

--- a/autopilot/hostfilter.go
+++ b/autopilot/hostfilter.go
@@ -195,35 +195,41 @@ func isUsableHost(cfg api.AutopilotConfig, gs api.GougingSettings, rs api.Redund
 // can be renewed, along with a list of reasons why it was deemed unusable.
 func isUsableContract(cfg api.AutopilotConfig, ci contractInfo, bh uint64, renterFunds types.Currency) (usable, refresh, renew, archive bool, reasons []error) {
 	c, s := ci.contract, ci.settings
-	if isOutOfCollateral(c, s, renterFunds, bh) {
-		reasons = append(reasons, errContractOutOfCollateral)
-		renew = false
-		refresh = true
-	}
-	if isOutOfFunds(cfg, s, c) {
-		reasons = append(reasons, errContractOutOfFunds)
-		renew = false
-		refresh = true
-	}
-	if shouldRenew, secondHalf := isUpForRenewal(cfg, c.Revision, bh); shouldRenew {
-		if secondHalf {
-			reasons = append(reasons, errContractUpForRenewal) // only unusable if in second half of renew window
-		}
-		renew = true
-		refresh = false
-	}
-	if c.Revision.RevisionNumber == math.MaxUint64 {
-		reasons = append(reasons, errContractMaxRevisionNumber)
-		archive = true // can't be revised anymore
-		renew = false
-		refresh = false
-	}
+
 	if bh > c.EndHeight() {
 		reasons = append(reasons, errContractExpired)
 		archive = true // expired
 		renew = false
 		refresh = false
 	}
+
+	if c.Revision.RevisionNumber == math.MaxUint64 {
+		reasons = append(reasons, errContractMaxRevisionNumber)
+		archive = true // can't be revised anymore
+		renew = false
+		refresh = false
+	}
+
+	if !archive {
+		if isOutOfCollateral(c, s, renterFunds, bh) {
+			reasons = append(reasons, errContractOutOfCollateral)
+			renew = false
+			refresh = true
+		}
+		if isOutOfFunds(cfg, s, c) {
+			reasons = append(reasons, errContractOutOfFunds)
+			renew = false
+			refresh = true
+		}
+		if shouldRenew, secondHalf := isUpForRenewal(cfg, c.Revision, bh); shouldRenew {
+			if secondHalf {
+				reasons = append(reasons, errContractUpForRenewal) // only unusable if in second half of renew window
+			}
+			renew = true
+			refresh = false
+		}
+	}
+
 	usable = len(reasons) == 0
 	return
 }

